### ISSUE TITLE
[Patch Request] Sixel support

### ISSUE THREAD

**Issue by smhmd** [bakkeby/st-flexipatch#7]
Referencing https://github.com/hackerb9/lsix/issues/32.

Base patch: https://gist.github.com/saitoha/70e0fdf22e3e8f63ce937c7f7da71809

Forks of patch in action:
- [charlesdaniels/st](https://github.com/charlesdaniels/st)
- [galatolofederico/st-sixel](https://github.com/galatolofederico/st-sixel)

**Comment by bakkeby** [comment]
Ah right, I remember seeing an email about this a long time ago, sorry for the profound delay.

It is a quite complicated thing to patch, but who's not up for a challenge?

For reference for anyone trying to go through this the first thing we'll want to do is to revert the following commit that removes the sixel stub code from st (because the other diffs depend on it).
https://git.suckless.org/st/commit/f74a9df6e1fc88eebe6d673d888b61fd83cf6fc4.html

Between the base patch and Charles Daniels' [st](https://github.com/charlesdaniels/st) I think I have managed to work out what changes are needed to st to make this work with v0.8.4.

It moves quite a bit of code from x.c to st.h, so I'll need some cross-patch testing to make sure this behaves as one would expect. At the very least I was able to get this to work with the alpha patch.

I'll be adding these changes on a branch called sixel for now until I have some confidence that it doesn't break too much.

**Comment by bakkeby** [comment]
To test run:
```gnuplot
$ gnuplot

        G N U P L O T
        Version 5.4 patchlevel 1    last modified 2020-12-01

        Copyright (C) 1986-1993, 1998, 2004, 2007-2020
        Thomas Williams, Colin Kelley and many others

        gnuplot home:     http://www.gnuplot.info
        faq, bugs, etc:   type "help FAQ"
        immediate help:   type "help"  (plot window: hit 'h')

Terminal type is now 'qt'
gnuplot> set terminal sixel

Terminal type is now 'sixelgd'
Options are 'nocrop enhanced size 640,480 font "arial,12.0" '
gnuplot> test
```

**Comment by bakkeby** [comment]
Interestingly the drawn sixel image is only visible against a dark background when combined with the alpha patch.
![image](https://user-images.githubusercontent.com/1438008/110674056-ccdb3c80-81d1-11eb-8b27-aecbf4124ce3.png)

**Comment by bakkeby** [comment]
Maybe that has something to do with that the sixel graphics are RBG and not RGBA.

**Comment by bakkeby** [comment]
The sixel support in itself is fine so I merged these changes and I raised #13 regarding the above transparency issue. I also added a note about this issue in the patches.def.h file.

I'll close this.
